### PR TITLE
media-gfx/iscan-plugin-gt-f720: version bumpt to 1.1.2 and EAPI=7

### PR DESCRIPTION
**Pull request by 4z0r**
The driver is downloaded from EPSON directly; in a bundle with the desired library.
Only the specific bundle is extracted. The drive has it's own ebuild.

Closes: https://bugs.gentoo.org/406425
Suggested-by: Kelzhadant <kelzhadant@gmail.com>
Signed-off-by: Ovidiu - Dan Bogat <4z0r@ovidiu.at>

### CODE REVIEW

**Comment by 4z0r** [comment]
Sources are fetched from EPSON directly:
http://download.ebz.epson.net/dsc/search/01/search/?OSC=LX

The Source however is a bundle with driver and the desired plugin (ebuild should only handle the plugin) which needed additional extraction.

Keywords haven been changed to "-* ~amd64 ~x86", since I can't test the plugin for functionality myself. 

See:
https://bugs.gentoo.org/406425

**Comment by 4z0r** [comment]
merged and squashed the PR #10292 within this one. 

Cleaned up the format of the commit message.

File: media-gfx/iscan-plugin-gt-f720/iscan-plugin-gt-f720-0.0.1.2-r1.ebuild [hunk 1]
```diff
@@ -0,0 +1,55 @@
+# Copyright 1999-2018 Gentoo Authors
```
Comment (mgorny) [inline media-gfx/iscan-plugin-gt-f720/iscan-plugin-gt-f720-0.0.1.2-r1.ebuild]: What's this ebuild doing? If you're doing a version bump, don't do unnecessary changes to the old version.

File: media-gfx/iscan-plugin-gt-f720/iscan-plugin-gt-f720-0.1.1.2.ebuild [hunk 2]
```diff
@@ -0,0 +1,58 @@
+# Copyright 1999-2018 Gentoo Foundation
```
Comment (mgorny) [inline media-gfx/iscan-plugin-gt-f720/iscan-plugin-gt-f720-0.1.1.2.ebuild]: Is it 1.1.2 or 0.1.1.2?

File: media-gfx/iscan-plugin-gt-f720/iscan-plugin-gt-f720-0.1.1.2.ebuild [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI="7"
+
+inherit multilib rpm
+
+DESCRIPTION="Epson Perfection V300 PHOTO scanner plugin for SANE 'epkowa' backend"
+HOMEPAGE="http://download.ebz.epson.net/dsc/search/01/search/?OSC=LX"
+SRC_URI="amd64? ( https://download2.ebz.epson.net/iscan/plugin/gt-f720/rpm/x64/iscan-gt-f720-bundle-1.0.1.x64.rpm.tar.gz -> ${PN}-bundle.rpm.tar.gz )
+	x86? ( https://download2.ebz.epson.net/iscan/plugin/gt-f720/rpm/x86/iscan-gt-f720-bundle-1.0.1.x86.rpm.tar.gz -> ${PN}-bundle.rpm.tar.gz )"
```
Comment (mgorny) [inline media-gfx/iscan-plugin-gt-f720/iscan-plugin-gt-f720-0.1.1.2.ebuild]: You're putting two different files into same file which is obviously wrong and will explode horribly on the other system you haven't tested. Also this is `1.0.1` now, so this ebuild is one horribly big WTF to me.

File: media-gfx/iscan-plugin-gt-f720/iscan-plugin-gt-f720-0.1.1.2.ebuild [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+DEPEND=">=media-gfx/iscan-2.21.0"
+RDEPEND="${DEPEND}"
+
+S="${WORKDIR}"
+
+QA_PREBUILT="usr/lib64/esci/libesci-interpreter-gt-f720.so*"
+
+src_unpack() {
+	unpack "${PN}-bundle.rpm.tar.gz"
+	mv * "./${PN}-bundle.rpm"
```
Comment (mgorny) [inline media-gfx/iscan-plugin-gt-f720/iscan-plugin-gt-f720-0.1.1.2.ebuild]: Don't do `*` guessing, use a conditional with the correct name instead.

File: media-gfx/iscan-plugin-gt-f720/iscan-plugin-gt-f720-0.1.1.2.ebuild [hunk 5]
```diff
@@ -0,0 +1,58 @@
...
+	srcrpm_unpack "./${PN}-bundle.rpm/plugins/esci-interpreter-gt-f720-0.1.1-2.x86_64.rpm"
+}
+
+src_install() {
+	local MY_LIB="/usr/$(get_libdir)"
```
Comment (mgorny) [inline media-gfx/iscan-plugin-gt-f720/iscan-plugin-gt-f720-0.1.1.2.ebuild]: You're using this variable exactly once. Kill it.

File: media-gfx/iscan-plugin-gt-f720/iscan-plugin-gt-f720-0.1.1.2.ebuild [hunk 6]
```diff
@@ -0,0 +1,58 @@
...
+
+	# install scanner firmware
+	insinto /usr/share/esci
+	doins "${WORKDIR}/usr/share/esci/"*
```
Comment (mgorny) [inline media-gfx/iscan-plugin-gt-f720/iscan-plugin-gt-f720-0.1.1.2.ebuild]: Why are you using absolute path to reference the current directory?

File: media-gfx/iscan-plugin-gt-f720/iscan-plugin-gt-f720-0.1.1.2.ebuild [hunk 7]
```diff
@@ -0,0 +1,58 @@
...
+
+	# install scanner plugins
+	insinto "${MY_LIB}/esci"
+	insopts -m0755
```
Comment (mgorny) [inline media-gfx/iscan-plugin-gt-f720/iscan-plugin-gt-f720-0.1.1.2.ebuild]: Sounds like you wanted `doexe`.

**Comment by a17r** [comment]
ping

**Comment by mgorny** [comment]
Closing due to no activity.
